ema/nextjs/src/components/dotcms-page.js

```diff
@@ -218,49 +215,97 @@ const Container = ({ containerRef }) => {
     // Get the contentlets for "this" container
     const contentlets = containers[identifier].contentlets[`uuid-${uuid}`];
 
+    // Memoize the contenlets to avoid re-rendering
+    const contentletsId = useMemo(() => {
+        return contentlets.map((contentlet) => contentlet.identifier);
+    }, [contentlets]);
+
+    // Memoize the page containers to avoid re-rendering
```

Comment:
You are not changing the state so not re-render will happened here...

We just need to get data out of the context... I think we just need a function that does this.

Reply:
Moved!